File: AndroidStuidoJava/app/src/main/java/jp/cha84rakanal/hellotogglecolor_studio_java/MainActivity.java
package jp.cha84rakanal.hellotogglecolor_studio_java;

import android.app.Activity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

public class MainActivity extends Activity {

    private boolean mToggle = true;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    void onButtonClick(View v){
        Log.v(MainActivity.class.getSimpleName(),"onButtonClick");
        mToggle = !mToggle;
        findViewById(R.id.back).setBackgroundColor( mToggle? 0xFFFF0000 : 0xFF0000FF);
    }
}

File: VisualStudioXamarine/HelloToggleColor_android/Droid/MainActivity.cs
using Android.App;
using Android.Widget;
using Android.OS;

namespace HelloToggleColor.Droid
{
    [Activity(Label = "HelloToggleColor", MainLauncher = true, Icon = "@mipmap/icon")]
    public class MainActivity : Activity
    {
        private bool toggle = true;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            // Set our view from the "main" layout resource
            SetContentView(Resource.Layout.Main);

            // Get our button from the layout resource,
            // and attach an event to it
            Button button = FindViewById<Button>(Resource.Id.myButton);

            button.Click += delegate {
                Android.Util.Log.Info("Xamarine","onClick");
                ((RelativeLayout)button.Parent).SetBackgroundColor((toggle)?Android.Graphics.Color.Red : Android.Graphics.Color.Blue);
                toggle = !toggle;
            };
        }
    }
}

